fusequery/query/src/optimizers/optimizer_constant_folding_test.rs

```diff
@@ -4,64 +4,111 @@
 
 #[cfg(test)]
 mod tests {
-    use std::mem::size_of;
-    use std::sync::Arc;
-
-    use common_datavalues::prelude::*;
     use common_exception::Result;
-    use common_planners::*;
-    use pretty_assertions::assert_eq;
 
-    use crate::optimizers::optimizer_test::*;
     use crate::optimizers::*;
 
     #[test]
     fn test_constant_folding_optimizer() -> Result<()> {
-        let ctx = crate::tests::try_create_context()?;
-
-        let total = ctx.get_settings().get_max_block_size()? as u64;
-        let statistics =
-            Statistics::new_exact(total as usize, ((total) * size_of::<u64>() as u64) as usize);
-        ctx.try_set_statistics(&statistics)?;
-        let source_plan = PlanNode::ReadSource(ReadDataSourcePlan {
-            db: "system".to_string(),
-            table: "test".to_string(),
-            schema: DataSchemaRefExt::create(vec![
-                DataField::new("a", DataType::Utf8, false),
-                DataField::new("b", DataType::Utf8, false),
-                DataField::new("c", DataType::Utf8, false),
-            ]),
-            parts: generate_partitions(8, total as u64),
-            statistics: statistics.clone(),
-            description: format!(
-                "(Read from system.{} table, Read Rows:{}, Read Bytes:{})",
-                "test".to_string(),
-                statistics.read_rows,
-                statistics.read_bytes
-            ),
-            scan_plan: Arc::new(ScanPlan::empty()),
-            remote: false,
-        });
-
-        let filter_plan = PlanBuilder::from(&source_plan)
-            .filter(col("a").gt(lit(6)).eq(lit(true)))?
-            .build()?;
+        #[allow(dead_code)]
+        struct Test {
+            name: &'static str,
+            query: &'static str,
+            expect: &'static str,
+        }
 
-        let plan = PlanNode::Projection(ProjectionPlan {
-            expr: vec![col("a")],
-            schema: DataSchemaRefExt::create(vec![DataField::new("a", DataType::Utf8, false)]),
-            input: Arc::from(filter_plan),
-        });
+        let tests: Vec<Test> = vec![
```

Comment:
Should we add more tests for boolean expressions, as mentioned in datafusion?

https://github.com/apache/arrow-datafusion/blob/f43dc444eb510f0ff170adecb5a23afb39c489a8/datafusion/src/optimizer/constant_folding.rs#L33

Reply:
This may be short circuit optimizer.

BTW: we can't optimize the short circuit in Kleene logic. because we can't determine whether null and bool. DataFuse uses Kleene logic.